File: Leetcode/python/sol53.py
# https://leetcode.com/problems/maximum-subarray/
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        max_endin = nums[0]
        max_current = nums[0]

        for i in range(1, len(nums)):
            max_ending = max(nums[i], max_ending + nums[i])
            max_current = max(max_ending, max_current)

        return max_current

File: Leetcode/cpp/sol53.cpp
// https://leetcode.com/problems/maximum-subarray/


class Solution {
public:
    int maxSubArray(vector<int>& nums) {

        int max_current = nums[0];
        int max_ending = nums[0];

        for (int i = 1; i < nums.size(); i++){
            max_ending = max(max_ending + nums[i], nums[i] );
            max_current = max(max_current, max_ending);
        }

        return max_current;
    }
};